File: cdk-from-cfn-testing/expected/sam_nodejs_lambda/typescript/SAMNodeJSLambdaConstruct.ts
import * as cdk from 'aws-cdk-lib';
import * as sam from 'aws-cdk-lib/aws-sam';
import { Construct } from 'constructs';

export interface SAMNodeJSLambdaConstructProps {
}

export class SAMNodeJSLambdaConstruct extends Construct {
  public constructor(scope: Construct, id: string, props: SAMNodeJSLambdaConstructProps = {}) {
    super(scope, id);

    // Transforms
    cdk.Stack.of(this).addTransform('AWS::Serverless-2016-10-31');

    // Resources
    const myFunction = new sam.CfnFunction(this, 'MyFunction', {
      runtime: 'nodejs20.x',
      handler: 'index.handler',
      inlineCode: 'exports.handler = async (event) => {\n  console.log(event);\n}\n',
    });
  }
}

File: cdk-from-cfn-testing/expected/sam_nodejs_lambda/csharp/SAMNodeJSLambdaConstruct.cs
using Amazon.CDK;
using Amazon.CDK.AWS.SAM;
using Constructs;
using System.Collections.Generic;

namespace SAMNodeJSLambdaConstruct
{
    public class SAMNodeJSLambdaConstructProps
    {
    }

    public class SAMNodeJSLambdaConstruct : Construct
    {
        public SAMNodeJSLambdaConstruct(Construct scope, string id, SAMNodeJSLambdaConstructProps props = null) : base(scope, id)
        {
            // Transforms
            Stack.Of(this).AddTransform("AWS::Serverless-2016-10-31");

            // Resources
            var myFunction = new CfnFunction(this, "MyFunction", new CfnFunctionProps
            {
                Runtime = "nodejs20.x",
                Handler = "index.handler",
                InlineCode = @"exports.handler = async (event) => {
                  console.log(event);
                }
                ",
            });
        }
    }
}
